cypress/components/badge/badge.cy.tsx

```diff
@@ -58,7 +56,7 @@ context("Testing Badge component", () => {
       CypressMountWithProviders(
         <BadgeComponent onClick={() => {}} counter="99" />
       );
-
+      // eslint-disable-next-line jest/valid-expect-in-promise
```

Comment:
why do we need this ?

Reply:
<img width="788" alt="Screenshot 2023-05-12 at 11 45 54" src="https://github.com/Sage/carbon/assets/88449494/9104083f-f48d-4e44-8329-a508127da7c5">


I was getting these errors while doing 'npm run commit' so Robin suggested me to add // eslint-disable-next-line jest/valid-expect-in-promise.